hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/validation/ValidatorRegistry.java

```diff
@@ -178,13 +251,8 @@ private EnumMap<RequestProcessingPhase, List<Method>> newPhaseMap() {
     return new EnumMap<>(RequestProcessingPhase.class);
   }
 
-  private <K, V> V getAndInitialize(K key, V defaultValue, Map<K, V> from) {
-    V inMapValue = from.get(key);
-    if (inMapValue == null || !from.containsKey(key)) {
-      from.put(key, defaultValue);
-      return defaultValue;
-    }
-    return inMapValue;
+  private <K, V> V getAndInitialize(K key, Supplier<V> defaultValue, Map<K, V> from) {
```

Comment:
Changed the function argument to a supplier to avoid redundant object declaration and gc for each and every call if found in the map.

Reply:
Already being done in https://github.com/apache/ozone/pull/6926